### PR TITLE
more veegee-like HE pipes (radiatiors)

### PR DESCRIPTION
**Pull request by CrazyAmphibian**
## What this does
changes the behavior of radiators
* radiators no longer heat transfer based on pressure change
* radiators no longer have an input and output side
* radiators have a new formula to determine convective heat transfer

## Why it's good
makes radiators less confusing to use
makes them operate more like /vg/ HE pipes

## How it was tested
went into game and put some gas in space to cool it, put some frezon in HE pipes in a room to see how it cooled it

**Changelog**
:cl:
- tweak: Changed how radiators worked

### CODE REVIEW

**Comment by CrazyAmphibian** [comment]
marked as draft so that the existing radiators are playtested and we see how people like them

**Review by brndd — CHANGES_REQUESTED** [review]
I'm not sure this implementation makes sense, or that this is a good change to begin with.

It looks like you're turning this into a unary device (only plugs in to anything on one end)? This would make it similar to the [thermal plate component](https://github.com/vgstation-coders/vgstation13/blob/d221546bcb17c8ac6ba2a0bb50ecb4f6c472e808/code/ATMOSPHERICS/components/unary/thermal_plate.dm) on /vg/13, rather than a [HE pipe](https://github.com/vgstation-coders/vgstation13/blob/d221546bcb17c8ac6ba2a0bb50ecb4f6c472e808/code/ATMOSPHERICS/he_pipes.dm). HE pipes on /vg/13 are part of a pipenet and have volume (so they contain gas), while thermal plates are unary components with no volume that transfer heat in and out of their attached pipenet.

The SS14 Heat Exchanger is a binary device that has gas move through it based on the pressure differential between its input and output -- essentially simulated flow. You'd use it by e.g. putting a pump on one or both ends. The gas that flows through it would then either lose or gain heat depending on whether it's hotter or colder than the environment.

This makes it different from /vg/13 HE pipes, which have no simulated flow through them (pipenets are just big containers, essentially) and cool down the contents of the entire pipenet instantaneously, which makes them unreasonably powerful in certain situations -- five dead end HE pipes in space connected directly to the station's main distro loop can supercool the entire station's distro in a few minutes, which makes no sense because (1) there should be no flow through those HE pipes, and (2) the surface area of those HE pipes is minuscule compared to the amount of gas the distro loop contains.

/vg/ HE pipes working like they do is a fundamentally broken design, because it breaks what should have been a rule of atmos pipes systems from the beginning: pipenets should be entirely passive systems ("containers"), and any changes to the temperature, volume, composition, etc. of a gas should *exclusively* happen in active devices. The fact that they break this rule is what leads to one of the major points of counterintuitiveness and stupidity of /vg/ pipes, which is the fact that cooling loops work even in the total absence of any gas flow, and due to the way our pumps work, adding flow in fact serves only to make the system *worse*.

So in short, I think this is not a good change for both technical and gameplay reasons:

- On a technical level, this implementation should be called the thermal plate and use its sprite to make it clear that it's a unary device.
- On a gameplay level, I feel like this drastically affects the balance of a gameplay system you (or me for that matter) don't yet understand very well, and likely for the worse. IMO /vg/ pipes are really not in a good place to begin with, so copying their gameplay 1:1 is a bad idea.

**Comment by CrazyAmphibian** [comment]
>It looks like you're turning this into a unary device

they're still (ostensibly) binary devices. 2 connections and whatnot.

> it breaks what should have been a rule of atmos pipes

well it's not, and i don't think that it needs to be since you always have the choice to not build with them - they're not the standard pipe, after all

> the fact that cooling loops work even in the total absence of any gas flow

mainly a flaw with pipes being a network system instead of per pipe


my main gripe with the system is that it adds nothing but extra steps without ever making the design require you to think any more than the old one. The total removal of passive systems is undeniably feature loss. It's a bit dumb, sure, but so is the game. I don't play ss1(3/4) to do fluid dynamics homework, i do it for funny spess man.
I'm sure some compromise could be made so that passive setups could be seen as "reliable, but not very effective", but the current implementation in both games does not allow for any middle ground.
